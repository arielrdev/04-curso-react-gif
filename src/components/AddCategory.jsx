import { useState } from "react"

export default function AddCategory() {

    const [inputValue, setInputValue] = useState('value initial')

    const onInputChange = (event) => {
        setInputValue(event.target.value)        
    }

    const onSubmit = (event) => {
        event.preventDefault();
        console.log(inputValue);
        
    }

    return (
        <form onSubmit={ onSubmit }>
            <input 
                type="text" 
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={ onInputChange }
            />
        </form>
  )
}
