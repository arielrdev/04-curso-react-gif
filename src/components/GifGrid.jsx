import { useEffect, useState } from "react";
import GifItem from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export default function GifGrid({ category }) {
    const { images, isLoading } = useFetchGifs ( category );
    console.log({ isLoading});

    return (
        <>
            <h3>{ category }</h3>
            <div className="card-grid">
                {
                    images.map( ( image ) => (
                        <GifItem 
                            key={image.id}
                            { ...image }
                            // title={image.title}
                            // ulr={image.title}
                        />
                    ))
                }
            </div>
        </>
    )
}
