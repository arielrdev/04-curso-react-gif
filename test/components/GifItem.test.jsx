import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('testing en <GifItem.jsx />', () => { 
    const title = 'Dragon Ball';
    const url = 'https://localhost/dragon.jpg';
    const alt = title;

    test('debe ser match con el snapshot', () => { 
        const { container } = render( <GifItem title={title} url={url} />)        
        expect( container ).toMatchSnapshot();
    });

    test('debe de mostar la img y title con la URL indicado', () => { 
        render(<GifItem title={title} url={ url } />)
        // expect( screen.getByRole('img').src).toBe( url );
        // expect( screen.getByRole('img').title).toBe( title );    
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe(url);
        expect( alt ).toBe( alt );
    });

    test('debe mostrar el titulo como un texto en el componente', () => { 
        render(<GifItem title={title} url={url} /> );
        expect( screen.getByText( title ) ).toBeTruthy();
     })
 })