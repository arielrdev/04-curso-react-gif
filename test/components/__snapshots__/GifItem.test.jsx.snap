// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing en <GifItem.jsx /> debe ser match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="https://localhost/dragon.jpg"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
